docs/content/2.components/form.md

```diff
@@ -117,16 +117,26 @@ You can manually set errors after form submission if required. To do this, simpl
 
 ```vue
 <script setup lang="ts">
-import type { FormError, FormSubmitEvent } from '#ui/types'
+import type { Form, FormSubmitEvent } from '#ui/types'
 
-const state = reactive({
+interface Schema {
+  email?: string
+  password?: string
+}
+
+const state = reactive<Schema>({
   email: undefined,
   password: undefined
 })
 
-const form = ref()
+const form = ref<Form<Schema>>()
+
+async function onSubmit (event: FormSubmitEvent<Schema>) {
+  if (!form.value) {
+    console.error("Form component not mounted yet")
+    return
+  }
```

Comment:
We don't really need this check this this function is bound to a client event.

```suggestion
```

Reply:
This was included to prevent the `'form.value' is possibly 'undefined'` error. Do you have any suggestions for alternative solutions?

Reply:
You can add a bang `!` operator like so: `form.value!.clear()`

Reply:
Alright, should be removed now